packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = queryString =>
+    document.querySelectorAll(queryString) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the queryString. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} queryString
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = queryString => {
```

Comment:
Since the term "query string" is very much tied to the URL, can we rename this variable `selector` or `querySelector`?